src/main/java/edu/carroll/dogtag/web/controller/UserProfileController.java

```diff
@@ -0,0 +1,87 @@
+package edu.carroll.dogtag.web.controller;
+
+import edu.carroll.dogtag.jpa.model.UserProfile;
+import edu.carroll.dogtag.service.LoginService;
+import edu.carroll.dogtag.service.UserProfileService;
+import edu.carroll.dogtag.web.form.UserProfileForm;
+import jakarta.servlet.http.HttpSession;
+import jakarta.validation.Valid;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.validation.BindingResult;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.ModelAttribute;
+import org.springframework.web.bind.annotation.PostMapping;
+
+/**
+ * The UserProfileController handles the information that is entered in the html pages to the UserProfileForm
+ * and allows the user to enter information for themselves into the database that is attached to
+ * their username that was registered. If not errors are found the information is sent to the registerService
+ * to handle the business logic before saving to the database.
+ */
+@Controller
+public class UserProfileController {
+    private static final Logger log = LoggerFactory.getLogger(UserProfileController.class);
+    private final UserProfileService userProfileService;
+    private final LoginService loginService;
+
+    public UserProfileController(UserProfileService userProfileService, LoginService loginService) {
+        this.userProfileService = userProfileService;
+        this.loginService = loginService;
+    }
+
+    /**
+     * @param model   used to add the TrainingForm to be able to pass it the PostMapping to
+     *                be used.
+     * @param session It allows the server to store and retrieve
+     *                user-specific data between requests.
+     * @return the UserProfileController is requested after the post of the button for submit is clicked
+     */
+    @GetMapping("/profilesetup")
+    public String profileForm(Model model, HttpSession session, String fname, String lname) {
+        final String user = (String) session.getAttribute("user");
+        if (user == null || user.isBlank()) {
+            log.error("Login user {} does not have session", user);
+            return "redirect:/login";
+        }
+        model.addAttribute("user", user);
+        model.addAttribute("userProfileForm", new UserProfileForm(user));
```

Comment:
Nit: If the user already has information in the database, you should probably fetch it and put it into the profile form, so they can update their information, rather than requiring them to re-enter it from scratch.